prow/github/types.go

```diff
@@ -346,6 +346,13 @@ type Repo struct {
 	Parent      ParentRepo      `json:"parent"`
 }
 
+// RepoName is a limited use version of Repo that holds only `name`
+// it is used in the v4 query to collect only names, this is faster than v3
+// See also https://docs.github.com/en/graphql/reference/objects#repository
+type RepoName struct {
```

Comment:
nit: Since this type is specific to the `GetRepoNamesFor*` functions I think it makes sense to colocate it with those functions and the rest of the related types. Unfortunately the types used for graphql queries are not typically as reusable as the REST API types.